### DISCUSSION TITLE
Is it possible to get ResourceInfo in a CustomSecurityIdentityAugmentor

### DISCUSSION THREAD

**Discussion by romario-mota** [quarkusio/quarkus#31069] [Q&A]
Hello,

I would like to obtain the ResourceInfo in a custom IdentityAugmentor, to perform the authorization in a personalized way, checking the annotations in the called method and manipulating the roles.

Is it possible to do that?

Thanks

[a comment by an automated account is not shown]

**Reply by sberyozkin** [reply]
@romario-mota Hi, do you mean JAX-RS `ResourceInfo` ? No, it is not possible.

What is possible is to have a JAX-RS `PreMatch` `ContainerRequestFilter` set a custom JAX-RS `SecurityContext` which then will be picked up by Quarkus Security. If it is to early in the pre-match phase for your case then perhaps the simplest is to use a post-match `ContainerRequestFilter`, set a custom SecurityContext there
